### PR TITLE
WIP Fix 362

### PR DESCRIPTION
**Pull request by nojaf**
@jindraivanek couple of problems with this solution:

I think `integrateComments` should receive the full configuration object.
Currently not possible due to FormatConfig and TokenMatcher being somewhat coupled.

The fix also looks a bit like firefighting, not sure if the problem is structurally fixed.

### CODE REVIEW

**Comment by jindraivanek** [comment]
> I think integrateComments should receive the full configuration object.

Agree.

> Currently not possible due to FormatConfig and TokenMatcher being somewhat coupled.

Please move `FormatConfig` type to separate file. This would also help us synchronizing config with Fantomas-UI.

File: src/Fantomas/FormatConfig.fs [hunk 1]
```diff
@@ -116,7 +116,7 @@ type internal ColumnIndentedTextWriter(tw : TextWriter) =
         member __.Dispose() =
             indentWriter.Dispose()    
 
-type internal Context = 
+type Context =
```
Comment (jindraivanek) [inline src/Fantomas/FormatConfig.fs]: It is removing `internal` really needed?

Comment (nojaf) [reply src/Fantomas/FormatConfig.fs]: No, my bad.

**Comment by nojaf** [comment]
@jindraivanek I think the extraction of Context.fs is not really that breaking to the outside world.
Would keep the new version as a revision, what do you think?

**Comment by jindraivanek** [comment]
Yeah, extraction of `Context` is better.

**Pull request merged**